src/map/quest.cpp

```diff
@@ -726,6 +726,12 @@ void quest_update_objective(struct map_session_data *sd, struct mob_data* md)
 					objective_check++;
 				if (qi->objectives[j]->mapid < 0 || (qi->objectives[j]->mapid == sd->bl.m && md->spawn != nullptr))
 					objective_check++;
+				else if (qi->objectives[j]->mapid >= 0) {
+					struct map_data *mapdata = map_getmapdata(sd->bl.m);
+
+					if (mapdata->instance_id && map_getmapdata(mapdata->instance_src_map)->m == qi->objectives[j]->mapid)
```

Comment:
```suggestion
					if (mapdata->instance_id && mapdata->instance_src_map == qi->objectives[j]->mapid)
```
This should already be the mapid.

Reply:
True!